crates/db/src/lib.rs

```diff
@@ -348,16 +359,23 @@ impl DB {
         Ok(res)
     }
 
-    pub async fn save_erc721_metadata(
+    pub async fn save_erc721_data(
         &self,
         address: Address,
         chain_id: u32,
         token_id: U256,
-        metadata: Erc721TokenMetadata,
+        owner: Address,
+        name: String,
+        symbol: String,
+        uri: String,
+        metadata: String,
     ) -> Result<()> {
         let mut conn = self.tx().await?;
+        queries::update_erc721(address, chain_id, token_id, owner, uri, metadata)
+            .execute(&mut conn)
+            .await?;
 
-        queries::update_erc721_metadata(address, chain_id, token_id, metadata)
+        queries::update_erc721collection(address, chain_id, name, symbol)
```

Comment:
these should be two separate calls: one to update a collection, and one to update a particular item. especially because they should happen with different frequencias (which is not how this currently does it)